src/vm/builtins/builtin_runner/builtin_runner.zig

```diff
@@ -156,4 +156,26 @@ pub const BuiltinRunner = union(enum) {
             .SegmentArena => .{ 0, 0 },
         };
     }
+
+    /// Retrieves the number of instances per component for the built-in runner.
+    ///
+    /// This function returns the number of instances per component for the specific type of
+    /// built-in runner. Each built-in runner may have a different number of instances per
+    /// component based on its configuration and purpose.
+    ///
+    /// # Returns
+    ///
+    /// A `usize` representing the number of instances per component for the built-in runner.
+    pub fn getInstancesPerComponent(self: *Self) u32 {
+        return switch (self.*) {
+            .Bitwise => |*bitwise| bitwise.instances_per_component,
+            .EcOp => |*ec| ec.instances_per_component,
+            .Hash => |*hash| hash.instances_per_component,
+            .RangeCheck => |*range_check| range_check.instances_per_component,
+            .Keccak => |*keccak| keccak.instances_per_component,
+            .Signature => |*signature| signature.instances_per_component,
+            .Poseidon => |*poseidon| poseidon.instances_per_component,
+            .SegmentArena, .Output => 1,
```

Comment:
```suggestion
            .SegmentArena, .Output => 1,
            else => |*builtin| builtin.instances_per_component
```

I am not sure which one would be better here